### PR TITLE
Simone matteo   skeleton links

### PR DESCRIPTION
**Pull request by chefleo**
Changes:

1. Implemented **skeleton loaders** in the 'Register' and 'Details' sections to improve user experience during loading times.
2. Added **Discord** and **GitHub** links for better user engagement and project visibility.
3. **Replaced CSS with image gradients** because the gradient doesn't show in particolar network (ex. Safari).

### CODE REVIEW

File: apps/next/src/pages/register.tsx [hunk 1]
```diff
@@ -27,11 +28,18 @@ export default function Result() {
           <Side_Navbar />
 
           {/* Register */}
-          <div className="flex-col w-full mt-9 pb-8 lg:mx-8 min-h-screen lg:min-h-full lg:w-3/4">
+          <div className="flex flex-col w-full mt-9 pb-8 lg:mx-8 min-h-screen lg:min-h-full lg:w-3/4 justify-between lg:justify-normal">
             {/* Three button Register, Details, Subdomain / Search Input (hidden mobile) */}
-            <Pages_buttons result={result} path={path} />
+            <div>
+              <Pages_buttons result={result} path={path} />
```
Comment (andrewgucci) [inline apps/next/src/pages/register.tsx:34]: I would suggest changing the component variable as well as its filename to `PageButtons`, `DomainOptionButtons`, or `DomainPages`. Having underscores in component variables and filenames looks tacky and the current name does not describe what this component is or does. I would also remove all underscores from other components like `Side_Navbar` and `Side_Navebar_Choise`, to keep things consistent across components.

File: apps/next/components/Search.tsx [hunk 2]
```diff
@@ -2,9 +2,17 @@ import React, { useState } from 'react'
 import { useRouter } from 'next/router'
 import Logo from '../public/Logo.png'
 import Search from '../public/Search.png'
+import Github from '../public/github.png'
```
Comment (andrewgucci) [inline apps/next/components/Search.tsx:5]: None of these imported logos `Github`, `Discord`, `Twitter`, etc. are used in this component and should ultimately only be required within the `Side_Navbar` or `Side_Navbar_Choice` component. I would eventually like to go through all files and remove unused, unnecessary, and redundant imports, but that can be a task for another pull request.

File: apps/next/components/Register/index.tsx [hunk 3]
```diff
@@ -197,43 +211,48 @@ export default function Register({ result }: { result: string }) {
           <div className="flex-col bg-gray-800 px-8 py-12 rounded-b-lg">
             <Alert
               available={
-                isNormalDomain && registerState !== RegisterState.Registered
+                isNormalDomain &&
+                registerState !== RegisterState.Registered &&
+                registerState !== undefined
```
Comment (andrewgucci) [inline apps/next/components/Register/index.tsx:216]: Why aren't we using an enum value to determine whether the domain is available? Why make the `registerState` variable `undefined` by default instead of setting it to a specific enum value like `Unregistered` to begin with or even `Registered` to begin with? Feels like we could reduce the number of boolean expressions throughout this component.

File: apps/next/components/Links.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      >
+        <div
+          className={`hover:scale-125 transform transition duration-400 ease-out`}
+        >
+          <Image
+            width={32}
+            height={32}
+            className="h-4 w-auto opacity-80 hover:opacity-100"
+            src={Twitter}
+            alt="FNS"
```
Comment (andrewgucci) [inline apps/next/components/Links.tsx:60]: If we're going to use alternative text for these images, the text should actually describe what the image is supposed to be. Please change the alt text for each to be specific for GitHub, Discord, Twitter, etc. not just "FNS" for all of them.

File: apps/next/components/Details/index.tsx [hunk 5]
```diff
@@ -29,7 +29,7 @@ export default function Details({ result }: { result: string }) {
 
   // State variable that changed inside Wagmi hooks
   const [prepared, setPrepared] = useState<boolean>(false)
-  const [isAvailable, setIsAvailable] = useState<boolean>(true)
+  const [isAvailable, setIsAvailable] = useState<boolean>()
```
Comment (andrewgucci) [inline apps/next/components/Details/index.tsx:32]: Why is `isAvailable` no longer `true` by default? Also wouldn't every domain default to `false` until availability has been confirmed with a contract read?

File: apps/next/components/Details/Info.tsx [hunk 6]
```diff
@@ -10,13 +10,13 @@ const InfoLine = ({
   alternativeText,
 }: {
   leftText: string
-  rightText: string
-  alternativeText: string
+  rightText: string | undefined;
```
Comment (andrewgucci) [inline apps/next/components/Details/Info.tsx:13]: Why would `rightText` ever be undefined? Wouldn't the default be `"Not Set"`?

File: apps/next/components/Register/index.tsx [hunk 7]
```diff
@@ -33,27 +33,41 @@ export enum RegisterState {
   Registered, // registration complete (count => 3)
 }
 
-const Alert = ({ available }: { available: boolean }) => {
+const Alert = ({ available, registerState }: { available: boolean; registerState:  RegisterState | undefined }) => {
```
Comment (andrewgucci) [inline apps/next/components/Register/index.tsx:36]: Is the register state undefined to show loading animations?

**Review by andrewgucci — CHANGES_REQUESTED** [review]
I'm sure I missed some stuff, but everything appears to be fairly solid. I'm primarily concerned with what a lot of the variable's default values are set to and how many boolean expressions are used throughout the components. I could also not understand the purpose so I primarily left a lot of questions related to my concerns and only a few direct changes.

File: apps/next/components/Register/index.tsx (hunk 3 above)
Comment (MatteoDigiorgio) [reply apps/next/components/Register/index.tsx:216]: If the ```registerState``` is set to either ```Unregistered``` or ```Registered```, then when the component is mounting, the page may briefly display an incorrect result for a fraction of a second. For example, if the actual domain availability is true but the default value of registerState in the code is ```Unregistered```, the page will briefly show that the domain is ```Unregistered``` and then, after the actual contract read, the page will show the correct state.

To avoid displaying an inaccurate state, even for a fraction of a second, I decided to use a skeleton animation. For this purpose, I needed to set the default value of registerState as follows: ```const [registerState, setRegisterState] = useState<RegisterState | undefined>();```

So, in short, if we allow the ```registerState``` to be ```undefined```, we can display a loading state while waiting for the correct value. Then, we can show whether the domain is registered or not once the correct value is obtained.

That being said, I actually think we can handle boolean expressions better by removing ```registerState !== undefined``` as it is unnecessary.

File: apps/next/components/Details/index.tsx (hunk 5 above)
Comment (MatteoDigiorgio) [reply apps/next/components/Details/index.tsx:32]: Same here. For the purpose of animation, I believe it is better to display a loading state before showing the actual value. From a user experience standpoint, it is not visually appealing to display that a domain is unavailable and then suddenly show that it is actually available. Additionally, if there is a situation with a poor internet connection, the user may see that the domain is unavailable for several seconds because the default value is displayed instead of the actual value. By handling this situation with a loading screen, I believe it provides a better experience for the user. Another, and even better, solution is to use an enum, as you suggested, to handle both situations we discussed. For example: 
```const State = {
  Registered: true,
  Unregistered: false,
  Unchecked: undefined,
} as const;```

File: apps/next/components/Details/Info.tsx (hunk 6 above)
Comment (MatteoDigiorgio) [reply apps/next/components/Details/Info.tsx:13]: Same reason here, it's for loading animation purpose.

File: apps/next/components/Register/index.tsx (hunk 7 above)
Comment (MatteoDigiorgio) [reply apps/next/components/Register/index.tsx:36]: Yes

File: apps/next/components/Register/index.tsx (hunk 3 above)
Comment (andrewgucci) [reply apps/next/components/Register/index.tsx:215]: Thank you for the clarification and examples Matteo, apologies if I came off ignorant or arrogant about the process. I now understand the use of `undefined` in the context of unmounted components and skeleton animations.

File: apps/next/components/Details/index.tsx (hunk 5 above)
Comment (andrewgucci) [reply apps/next/components/Details/index.tsx]: I would agree that having loading states before actual information has been read or loaded is better for the user experience. Thank you again for the thoughtful and detailed explanation.

**Review by andrewgucci — APPROVED** [review]
Everything looks very clean and consistent, beautiful work as always. Thank you @MatteoDigiorgio for resolving my confusion with grace and helping me understand what was happening in the components. Approved!
